test/hotspot/jtreg/compiler/exceptions/OptimizeImplicitExceptions.java

```diff
@@ -0,0 +1,411 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+ /*
+ * @test
+ * @bug 8273563
+ * @summary Test -XX:+/-OmitStackTraceInFastThrow and -XX:+/-OptimizeImplicitExceptions
+ *
+ * @requires vm.compiler2.enabled & vm.compMode != "Xcomp"
+ *
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI
+ *                   -XX:+UseSerialGC -Xbatch -XX:-UseOnStackReplacement -XX:-TieredCompilation
+ *                   -XX:CompileCommand=compileonly,compiler.exceptions.OptimizeImplicitExceptions::throwImplicitException
+ *                   compiler.exceptions.OptimizeImplicitExceptions
+ */
+
+package compiler.exceptions;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+
+import jdk.test.lib.Asserts;
+import jdk.test.whitebox.WhiteBox;
+
+public class OptimizeImplicitExceptions {
+    // ImplicitException represents the various implicit (aka. 'built-in') exceptions
+    // which can be thrown implicitely by the JVM when executing bytecodes.
+    public enum ImplicitException {
+        // NullPointerException during field access
+        NULL_POINTER_EXCEPTION("null_check"),
+        // NullPointerException during invoke
+        INVOKE_NULL_POINTER_EXCEPTION("null_check"),
+        ARITHMETIC_EXCEPTION("div0_check"),
+        ARRAY_INDEX_OUT_OF_BOUNDS_EXCEPTION("range_check"),
+        ARRAY_STORE_EXCEPTION("array_check"),
+        CLASS_CAST_EXCEPTION("class_check");
+        private final String reason;
+        ImplicitException(String reason) {
+            this.reason = reason;
+        }
+        public String getReason() {
+            return reason;
+        }
+    }
+    // TestMode represents a specific combination of the OmitStackTraceInFastThrow/OptimizeImplicitExceptions
+    // command line options. They will be set up in 'setFlags(TestMode testMode)' before a new test run starts.
+    public enum TestMode {
+        OMIT_STACKTRACES_IN_FASTTHROW,
+        STACKTRACES_IN_FASTTHROW,
+        STACKTRACES_IN_FASTTHROW_OPTIMIZED
+    }
+
+    private static final WhiteBox WB = WhiteBox.getWhiteBox();
+    // Until JDK-8275908 is not fixed, null-pointer traps for invokes and array-store traps are not profiled in the interpreter.
+    private static final boolean JDK8275908_fixed = false;
```

Comment:
I don't know if that one should get fixed first, but I'm ok with your workaround. Would it make sense to add that bug id to this test's header?

Reply:
This PR is now open for so long time and I'd like to complete it without the dependency on another fix. But adding the bug id to the test is a good idea. Done